site/en/tutorials/images/data_augmentation.ipynb

```diff
@@ -377,44 +486,98 @@
       "metadata": {
         "colab": {},
         "colab_type": "code",
-        "id": "fvgz_6t21dq2"
+        "id": "r1Bt7w5VhVDY"
       },
       "outputs": [],
       "source": [
-        "cropped = tf.image.central_crop(image, central_fraction=0.5)\n",
-        "visualize(image,cropped)"
+        "aug_ds = train_ds.map(\n",
+        "  lambda x, y: (resize_and_rescale(x, training=True), y))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "HKqeahG2hVdV"
+      },
+      "source": [
+        "With this approach, you use `Dataset.map` to create a dataset that yields batches of augmented images. In this case:\n",
+        "\n",
+        "* Data augmentation will happen asynchronously on the CPU, and is non-blocking. You can overlap the training of your model on the GPU with data preprocessing, using `Dataset.prefetch`, shown below.\n",
+        "* In this case, you will need to reimplement preprocessing logic server-side, as the prepreprocessing layers will not be exported with the model when you call `model.save`.\n"
```

Comment:
Yes, but I would also mention that after training you can attach the preprocessing before export.